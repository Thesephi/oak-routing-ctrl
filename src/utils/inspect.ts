export const inspect = (obj: unknown, opts?: object): string =>
  globalThis.Deno ? Deno.inspect(obj, opts) : JSON.stringify(obj, undefined, 2);
