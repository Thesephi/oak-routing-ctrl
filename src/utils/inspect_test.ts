import { assertSnapshot } from "@std/testing/snapshot";
import { inspect } from "./inspect.ts";

const testObj = {
  this: {
    is: {
      a: ["relatively", "deep", "object"],
      b: {
        but: {
          "is it": true,
        },
        or: {
          is: {
            it: new Map([["just", "1"], ["a", "2"], ["hoax", "3"]]),
          },
        },
      },
    },
  },
};

Deno.test("inspect without depth - Deno env", async (t) => {
  const retVal = inspect(testObj);
  await assertSnapshot(t, retVal);
});

Deno.test("inspect with depth - Deno env", async (t) => {
  const retVal = inspect(testObj, { depth: 10 });
  await assertSnapshot(t, retVal);
});

Deno.test("inspect - non-Deno env", async (t) => {
  const origDesc = Object.getOwnPropertyDescriptor(globalThis, "Deno");
  Object.defineProperty(globalThis, "Deno", {
    value: undefined,
  });
  const retVal = inspect(testObj);
  Object.defineProperty(globalThis, "Deno", origDesc!);
  await assertSnapshot(t, retVal);
});
