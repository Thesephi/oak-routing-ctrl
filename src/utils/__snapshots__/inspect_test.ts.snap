export const snapshot = {};

snapshot[`inspect without depth - Deno env 1`] = `
'{
  this: {
    is: {
      a: [ "relatively", "deep", "object" ],
      b: { but: { "is it": true }, or: { is: [Object] } }
    }
  }
}'
`;

snapshot[`inspect with depth - Deno env 1`] = `
'{
  this: {
    is: {
      a: [ "relatively", "deep", "object" ],
      b: {
        but: { "is it": true },
        or: {
          is: { it: Map(3) { "just" => "1", "a" => "2", "hoax" => "3" } }
        }
      }
    }
  }
}'
`;

snapshot[`inspect - non-Deno env 1`] = `
'{
  "this": {
    "is": {
      "a": [
        "relatively",
        "deep",
        "object"
      ],
      "b": {
        "but": {
          "is it": true
        },
        "or": {
          "is": {
            "it": {}
          }
        }
      }
    }
  }
}'
`;
